File: November/Week1/SoHyun/S3_18429.py
def find(x, d):
    global result
    if x < 500:
        return
    if d == n:
        result += 1
        return
    x -= k
    
    for i in range(n):
        if not v[i]:
            v[i] = 1
            find(x + kit[i], d+1)
            v[i] = 0

n, k = map(int, input().split())
kit = list(map(int, input().split()))

result = 0
v = [0] * n

find(500, 0)
print(result)

File: November/Week1/Semin/S3_18429.java
import java.util.*;
import java.io.*;

class Main {
    static int N;
    static int K;
    static int[] kit;
    static boolean[] visited;
    static int result = 0;
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] NK = br.readLine().split(" ");
        N = Integer.parseInt(NK[0]);
        K = Integer.parseInt(NK[1]);
        visited = new boolean[N];
        kit = new int[N];
        
        String[] s = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            kit[i] = Integer.parseInt(s[i]);
        }  

        back(0, 500);
        System.out.println(result);
    }
    static void back(int depth, int weight){
        if(depth == N){
            if(weight >= 500){
                result++;
                return;
            }
        }
        for(int i = 0; i < N; i++){
            if(!visited[i] && weight + kit[i] - K >= 500){
                visited[i] = true;
                back(depth + 1, weight + kit[i] - K);
                visited[i] = false;
            }
        }
    }
    
}